File: ueueo-framework/ueueo-blob-storing/src/main/java/com/ueueo/BlobStoring/DefaultBlobContainerConfigurationProvider.cs
using Microsoft.Extensions.Options;
using Volo.Abp.DependencyInjection;

namespace Volo.Abp.BlobStoring;

public class DefaultBlobContainerConfigurationProvider : IBlobContainerConfigurationProvider, ITransientDependency
{
    protected AbpBlobStoringOptions Options;//  { get; }

    public DefaultBlobContainerConfigurationProvider(IOptions<AbpBlobStoringOptions> options)
    {
        Options = options.Value;
    }

    public   BlobContainerConfiguration Get(String name)
    {
        return Options.Containers.GetConfiguration(name);
    }
}

File: ueueo-framework/ueueo-blob-storing/src/main/java/com/ueueo/blobstoring/DefaultBlobContainerConfigurationProvider.java
package com.ueueo.blobstoring;

public class DefaultBlobContainerConfigurationProvider implements IBlobContainerConfigurationProvider {
    protected AbpBlobStoringOptions options;

    public DefaultBlobContainerConfigurationProvider(AbpBlobStoringOptions options) {
        this.options = options;
    }

    @Override
    public BlobContainerConfiguration get(String name) {
        return options.getContainers().getConfiguration(name);
    }

    public AbpBlobStoringOptions getOptions() {
        return options;
    }
}
